osu.Game/Database/LegacyModelExporter.cs

```diff
@@ -0,0 +1,164 @@
+        /// <param name="realm">The RealmAccess used to provide the exported file.</param>
+        protected LegacyModelExporter(Storage storage, RealmAccess realm)
+        {
+            exportStorage = storage.GetStorageForDirectory(@"exports");
+            UserFileStorage = storage.GetStorageForDirectory(@"files");
+            realmAccess = realm;
+        }
+
+        /// <summary>
+        /// Export the model to default folder.
+        /// </summary>
+        /// <param name="model">The model should export.</param>
+        /// <param name="cancellationToken">
+        /// The Cancellation token that can cancel the exporting.
+        /// If specified CancellationToken, then use it. Otherwise use PostNotification's CancellationToken.
+        /// </param>
+        /// <returns></returns>
+        public async Task ExportAsync(TModel model, CancellationToken? cancellationToken = null)
+        {
+            // check if the model is being exporting already
+            if (!exportingModels.Contains(model))
+            {
+                exportingModels.Add(model);
+            }
+            else
+            {
+                PostNotification?.Invoke(new SimpleErrorNotification
+                {
+                    Text = "File is being exported"
+                });
+                return;
+            }
+
+            string itemFilename = GetFilename(model).GetValidFilename();
+            IEnumerable<string> existingExports =
+                exportStorage
+                    .GetFiles(string.Empty, $"{itemFilename}*{FileExtension}")
+                    .Concat(exportStorage.GetDirectories(string.Empty));
+            string filename = NamingUtils.GetNextBestFilename(existingExports, $"{itemFilename}{FileExtension}");
+            bool success = false;
+
+            ProgressNotification notification = new ProgressNotification
+            {
+                State = ProgressNotificationState.Active,
+                Text = "Exporting...",
+                CompletionText = "Export completed"
+            };
+            notification.CompletionClickAction += () => exportStorage.PresentFileExternally(filename);
+            PostNotification?.Invoke(notification);
+
+            try
+            {
+                using (var stream = exportStorage.CreateFileSafely(filename))
+                {
+                    success = await ExportToStreamAsync(model, stream, notification, cancellationToken ?? notification.CancellationToken).ConfigureAwait(false);
+                }
+            }
+            catch (OperationCanceledException)
+            {
+                success = false;
+            }
+            finally
+            {
+                // cleanup if export is failed or canceled.
+                if (!success)
+                {
+                    exportStorage.Delete(filename);
+                }
```

Comment:
Why not move this outside of the finally and `return` in the `catch`?

Reply:
If it is not finally, the exported error (An error other than `OperationCanceledException` has occurred) file should not be deleted.
If it is determined that keeping the wrong file is the expected behavior, then I will change it,